lava/lava-job-definitions/shared/templates/base.yaml

```diff
@@ -89,3 +89,14 @@ notify:
       status: finished
     verbosity: verbose
 {% endif %}
+
+{% if callback_domain and callback_port %}
+ notify:
+    criteria:
+       status: finished
+    callbacks:
+      - url: http://{{ callback_domain }:{{ callback_port }}/
+        method: POST
+        dataset: minimal
```

Comment:
Why is it minimal? According to documentation minimal is "This will provide basic job info such as job id, status, submit_time, start_time, end_time, submitter_username, failure_comment, priority, description, actual_device_id, definition and metadata."
I think we would need "results": "In addition to minimal data this will also attach the job results as a list of test suites exported in yaml format."

What do you think?

Reply:
Currently it add quite a lot of unnecessary data (f.ex. logs) which count more than 95% of the payload. Same data can be queries using xmlrpc.

Reply:
but I guess we need "results", otherwise how do we know which tests passed or failed?

Reply:
Yes, https://github.com/ARMmbed/sandbox/blob/ollvai-lava/salt/states/lava-master/files/data-proxy/server.py current server implementation uses xmlrpc to fetch the test suite results. I'd like to keep it this way, so that the same code can be used for fetching the results later, if server f.ex crashes. In that case user could trigger the data fetching just be giving the job id (request post would not depend on having the result data)

Reply:
Ok, fair enough.